biloba/service.py

```diff
@@ -143,7 +139,7 @@ def stop(self, block=True):
         if not self.started:
             return
 
-        self._kill.set()
+        self.pool.kill(block=block)
```

Comment:
I don't think this is the point at which to block when calling stop - let ``_run_thread.get`` do that.